portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,26 +44,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         Query query = new Query("Comment");
         PreparedQuery results = datastore.prepare(query);
         
-        List<String> texts = new ArrayList<>();
+        List<String[]> cmts = new ArrayList<>();
         for (Entity entity : results.asIterable()) {
-            String text = (String) entity.getProperty("text");
-            texts.add(text);    
+            String[] pair = {(String) entity.getProperty("email"),
+                             (String) entity.getProperty("text")};
+            cmts.add(pair);    
         }
         
         Gson gson = new Gson();
-        String json = gson.toJson(texts);
+        String json = gson.toJson(cmts);
         response.setContentType("application/json;");
         response.getWriter().println(json);
+        response.sendRedirect("/index.html");
```

Comment:
This line is causing the JSON syntax error in your Javascript. And that should be the case, because redirecting to `/index.html` will overwrite the response with the content of the `/index.html`, which is an HTML that matches the JSON syntax error (unrecognized JSON token `<`). Not sure why you added this line here in the first place... Ah, probably because you weren't able to redirect correctly after login/logout, and Alec helped fixing that issue while we were meeting tonight. So now let's remove this line and see if it works (I'm pretty sure it will).